### PR TITLE
WebcalRefreshJob: Fix reading refresh rate

### LINKED ISSUE #19146: Calendar events from subscribed calendars do not get updated
### Steps to reproduce
1. Subscribe to ToDoist (Settings, integrations, calendar subscription url) 
2. Add new subscription via this url.
3. After next cron job run the events are displayed
4. Change the due date of a task in todoist
5. Wait after cron job has run

### Expected behaviour
The todoist event should have been moved to the new due date

### Actual behaviour
Event remains unchanged.

Cron jobs are running.
Calling the url manually delivers the new due date:

BEGIN:VEVENT
SUMMARY:* Bilanz 2018
UID:4357847-3540498040@todoist.com
DTSTART;VALUE=DATE:**20200131**
DTEND;VALUE=DATE:20200201
DESCRIPTION:Projekt: Buchhaltung Arguna / Petra\n\n\nDiesen Eintrag erledigen: \nhttps://todoist.com/showTask?sync_id=3622798163&id=3540498040
END:VEVENT

The row in calenderobjects still has the old due date in calenderdata:

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Sabre//Sabre VObject 4.1.6//EN
CALSCALE:GREGORIAN
BEGIN:VEVENT
SUMMARY:* Bilanz 2018
UID:4357847-3540498040@todoist.com
DTSTART;VALUE=DATE:**20200124**
DTEND;VALUE=DATE:20200125
DESCRIPTION:Projekt: Buchhaltung Arguna / Petra\n\n\nDiesen Eintrag erledig
 en: \nhttps://todoist.com/showTask?sync_id=3622798163&id=3540498040
END:VEVENT
END:VCALENDAR

So the initial loading of events after subscription does work, but the events do not get updated. 

This happens always with todoist subscriptions, I can reproduce the behavior by deleting the subscription and adding it again. Also happens with todoist project subscriptions, which show only the events from a specific project.

This is a regression, it did work with the calendar app version before 2.0.0.     

### Calendar app
**Calendar app version:** 2.0.1

**CalDAV-clients used:** DAVx5

### Client configuration
**Browser:** Chromium and Firefox

**Operating system:** Ubuntu 19.10

### Server configuration

**Operating system**: FreeBSD 11.3

**Web server:** Caddy 1.0.4

**Database:**  MariaDB

**PHP version:** 7.3

**Nextcloud Version:** 17.0.2

**Updated from an older installed version or fresh install:**

Updated from previous version

**List of activated apps:**

See nextcloud/calendar#1899 from yesterday

**Nextcloud configuration:**

See nextcloud/calendar#1899

#### Web server error log (e.g. /var/log/apache)

See nextcloud/calendar#1899 

#### Log file (data/nextcloud.log)

See nextcloud/calendar#1899

#### Browser log

See nextcloud/calendar#1899

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #19146

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Looks good

File: apps/dav/lib/BackgroundJob/RefreshWebcalJob.php [hunk 1]
```diff
@@ -461,4 +468,25 @@ private function cleanURL(string $url) {
...
 		return $cleanURL;
 	}
+
+	/**
+	 * Fixes types of rows
+	 *
+	 * @param array $row
+	 */
+	private function fixSubscriptionRowTyping(array &$row):void {
+		$forceInt = [
```
Comment (tcitworld) [inline apps/dav/lib/BackgroundJob/RefreshWebcalJob.php]: `const`?

**Review by tcitworld — APPROVED** [review]

**Comment by dev4223** [comment]
I added the fix in https://github.com/nextcloud/server/pull/19228 to my Nextcloud installation (18.0.4 with Calendar app 2.0.2) and it didn't work. Google calendar events shown after subscription but the calendar is not getting updated after this.

**Comment by panfanky** [comment]
In 20.0.4 to make the surscribed calendars refresh, I had to go to MySQL table `oc_calendarsubscriptions` and change NULL at refreshrate to `PT1H` as mentioned in [#19146](https://github.com/nextcloud/server/issues/19146).

**Pull request merged**